arcjet-bun/index.ts

```diff
@@ -0,0 +1,229 @@
+import { createConnectTransport } from "@connectrpc/connect-node";
+import core, {
+  ArcjetDecision,
+  ArcjetOptions,
+  Primitive,
+  Product,
+  ArcjetHeaders,
+  Runtime,
+  ArcjetRequest,
+  ExtraProps,
+  RemoteClient,
+  RemoteClientOptions,
+  defaultBaseUrl,
+  createRemoteClient,
+  Arcjet,
+} from "arcjet";
+import findIP from "@arcjet/ip";
+import type { Server } from "bun";
+
+// Re-export all named exports from the generic SDK
+export * from "arcjet";
+
+// Type helpers from https://github.com/sindresorhus/type-fest but adjusted for
+// our use.
+//
+// Simplify:
+// https://github.com/sindresorhus/type-fest/blob/964466c9d59c711da57a5297ad954c13132a0001/source/simplify.d.ts
+// EmptyObject:
+// https://github.com/sindresorhus/type-fest/blob/b9723d4785f01f8d2487c09ee5871a1f615781aa/source/empty-object.d.ts
+//
+// Licensed: MIT License Copyright (c) Sindre Sorhus <sindresorhus@gmail.com>
+// (https://sindresorhus.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions: The above copyright
+// notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+type Simplify<T> = { [KeyType in keyof T]: T[KeyType] } & {};
+declare const emptyObjectSymbol: unique symbol;
+type WithoutCustomProps = {
+  [emptyObjectSymbol]?: never;
+};
+
+type PlainObject = {
+  [key: string]: unknown;
+};
+
+export function createBunRemoteClient(
+  options?: RemoteClientOptions,
+): RemoteClient {
+  // The base URL for the Arcjet API. Will default to the standard production
+  // API unless environment variable `ARCJET_BASE_URL` is set.
+  const baseUrl = options?.baseUrl ?? defaultBaseUrl();
+
+  // Transport is the HTTP client that the client uses to make requests.
+  const transport =
+    options?.transport ??
+    createConnectTransport({
+      baseUrl,
+      httpVersion: "1.1",
```

Comment:
Is switching this to `2` blocked by https://github.com/arcjet/arcjet/issues/1180 ?

Reply:
Yep